packages/medusa/src/services/product-variant.ts

```diff
@@ -349,7 +359,9 @@ class ProductVariantService extends TransactionBaseService {
 
       for (const price of prices) {
         if (price.region_id) {
-          const region = await this.regionService_.retrieve(price.region_id)
+          const region = await this.regionService_
```

Comment:
todo: move that outside of the loop `const regionsServiceTx = this.regionService_.withTransaction(manager)`